src/coreclr/jit/lclvars.cpp

```diff
@@ -5872,6 +5872,14 @@ void Compiler::lvaFixVirtualFrameOffsets()
     for (TempDsc* temp = codeGen->regSet.tmpListBeg(); temp != nullptr; temp = codeGen->regSet.tmpListNxt(temp))
     {
         temp->tdAdjustTempOffs(delta);
+#if defined(TARGET_ARM64)
```

Comment:
Is there some guarantee that all the predicate temps end up adjacent on this list? Otherwise it seems like this indexing scheme might not work out.

Reply:
If you see below, we iterate over all the `type` and call `tmpPreAllocateTemps` with the number of slots we need for that type.

https://github.com/dotnet/runtime/blob/2b4a4bcfcaacd3614936c4cddeee438907bfd780/src/coreclr/jit/lsra.cpp#L7814-L7828

In `tmpPreAllocateTemps()`, we iterate through the number of slots we want to allocate and create them:

https://github.com/dotnet/runtime/blob/2b4a4bcfcaacd3614936c4cddeee438907bfd780/src/coreclr/jit/regset.cpp#L694-L708